src/main/java/javax/jmdns/impl/JmDNSImpl.java

```diff
@@ -1018,12 +1018,15 @@ public void registerService(ServiceInfo infoAbstract) throws IOException {
         info.addAddress(_localHost.getInet4Address());
         info.addAddress(_localHost.getInet6Address());
 
+        this.waitForAnnounced(DNSConstants.SERVICE_INFO_TIMEOUT);
```

Comment:
How is this related to the bug?
Isn't this dangerous here as it blocks the method from returning?

Reply:
Sorry, I will have to take a look to verify why is this shown as it is part of my commit. It is not. I presume it was there before my changes and I just moved it.

Reply:
I did a quick check and this strings appears in two commits
* 9689c79aa270572a1f48878a52a93ccbe4aa3d9a -  Added timeout on waitForAnnounced
* fee0d4c27cdc159a1b4a0fa5883ddf562a696d00 -  Remove waits. Fixes #144 (#174)

I'm not sure why those waits got back in my commit, but I will remove it. This is the only non intended difference between my commit and fee0d4c27cdc159a1b4a0fa5883ddf562a696d00 As far as I remember this happened in some intermediate release.